examples/showcase/tracked-controls/js/components/collider.js

```diff
@@ -0,0 +1,109 @@
+/* global AFRAME, THREE */
+
+/**
+ * It implements AABB collision detection for entities with a mesh.
+ * (https://en.wikipedia.org/wiki/Minimum_bounding_box#Axis-aligned_minimum_bounding_box)
+ * It sets the specified state on the intersected entities.
+ *
+ * @property {string} Selector of the entities to test for collision.
+ * @property {string} State to set on collided entities.
+ *
+ */
+AFRAME.registerComponent('collider', {
+  schema: {
+    objects: {default: ''},
+    state: {default: 'collided'}
+  },
+
+  init: function () {
+    this.els = [];
+    this.collisions = [];
+    this.elMax = new THREE.Vector3();
+    this.elMin = new THREE.Vector3();
+  },
+
+  update: function () {
+    this.updateEls();
+  },
+
+  /**
+   * Update list of entities to test for collision.
+   */
+  updateEls: function () {
+    var data = this.data;
+    var objectEls;
+
+    // Push entities into list of els to intersect.
+    if (data.objects) {
+      objectEls = this.el.sceneEl.querySelectorAll(data.objects);
+    } else {
+      // If objects not defined, intersect with everything.
+      objectEls = this.el.sceneEl.children;
+    }
+    // Convert from NodeList to Array
+    this.els = Array.prototype.slice.call(objectEls);
+  },
+
+  tick: (function () {
+    var boundingBox = new THREE.Box3();
```

Comment:
can we do `this.boundingBoxHelper = new THREE.Box3()` in `init` should we don't have to wrap?